cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -415,26 +444,24 @@ private void restartIfNecessary(NodeRef nodeRef, RestartContext restartContext)
         restartContext.restartReasons = podNeedsRestart.apply(pod);
 
         try {
-            checkReconfigurability(nodeRef, pod, restartContext);
+            checkIfRestartOrReconfigureRequired(nodeRef, restartContext);
             if (restartContext.forceRestart) {
                 LOGGER.debugCr(reconciliation, "Pod {} can be rolled now", nodeRef);
                 restartAndAwaitReadiness(pod, operationTimeoutMs, TimeUnit.MILLISECONDS, restartContext);
             } else if (restartContext.needsRestart || restartContext.needsReconfig) {
                 if (deferController(nodeRef, restartContext)) {
-                    LOGGER.debugCr(reconciliation, "Pod {} is controller and there are other pods to verify. Non-controller pods will be verified first.", nodeRef);
-                    throw new ForceableProblem("Pod " + nodeRef.podName() + " is controller and there are other pods to verify. Non-controller pods will be verified first");
+                    LOGGER.debugCr(reconciliation, "Pod {} is the active controller and there are other pods to verify. Other pods will be verified first.", nodeRef);
+                    throw new ForceableProblem("Pod " + nodeRef.podName() + " is the active controller and there are other pods to verify. Other pods will be verified first.");
```

Comment:
I have addressed the first point. The second point, as you pointed out, is not caused by this PR therefore leaving it.

Reply:
Yes, I have already addressed that part. So the error message is returned without `.`.  

> Because if other pods are rolled, it will not be really retried after 250ms, but not sure how much this is caused by this PR, so feel free to ignore this is not.

This is the part I'm leaving for now as you suggested.